### ISSUE TITLE
Writing triangle mesh into ply files with arbitrary data types

### ISSUE THREAD

**Issue by Tofull** [isl-org/Open3D#4589]
We can write pointclouds in ply files with arbitrary data types (see quote below), including float32, but if we have a mesh derivated from this pointcloud, it will be written as a legacy mesh, so we are stuck to double precision.

> Writing ply files with arbitrary data types is supported in the tensor based point cloud IO:
http://www.open3d.org/docs/latest/python_api/open3d.t.io.write_point_cloud.html#open3d.t.io.write_point_cloud
>
> Please test and let us know if this works for you.
>
> _Originally posted by @ssheorey in https://github.com/isl-org/Open3D/issues/1325#issuecomment-840827943_

What about writing `open3d.t.geometry.TriangleMesh` in arbitrary data types (especially float32) ?

**Comment by reyanshsolis** [comment]
Yes, you are right. Currently `TriangleMesh` does not have a `native` `IO` for `t` or `tensor-pipeline`, it is converted to and from legacy for IO operation. We might be adding this in the future. In case you are interested in contributing, I can help you with the same. 
Thanks
